src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -6295,29 +6302,46 @@ internal void TdsLogin(SqlLogin rec, TdsEnums.FeatureExtension requestedFeatures
             _physicalStateObj._messageStatus = 0;
         }// tdsLogin
 
+#if MANAGED_SNI
+        private void SSPIData(byte[] receivedBuff, UInt32 receivedLength, ref byte[] sendBuff, ref UInt32 sendLength)
+        {
+            SNISSPIData(receivedBuff, receivedLength, ref sendBuff, ref sendLength);
+        }
+#else
         private void SSPIData(byte[] receivedBuff, UInt32 receivedLength, byte[] sendBuff, ref UInt32 sendLength)
         {
             SNISSPIData(receivedBuff, receivedLength, sendBuff, ref sendLength);
         }
+#endif
+
 
+#if MANAGED_SNI
+        private void SNISSPIData(byte[] receivedBuff, UInt32 receivedLength, ref byte[] sendBuff, ref UInt32 sendLength)
+        {
+            uint clientContextResult = SNIProxy.Singleton.GenSspiClientContext(_physicalStateObj.Handle, receivedBuff, ref sendBuff, _sniSpnBuffer);
+            if (clientContextResult != SNIProxy.SspiClientContextResult.OK)
+            {
+                string errorMessage = clientContextResult == SNIProxy.SspiClientContextResult.KerberosTicketMissing ?
+                                        "Kerberos Ticket is missing. Run 'kinit'." : SQLMessage.SSPIGenerateError();
+                SSPIError(errorMessage, TdsEnums.GEN_CLIENT_CONTEXT);
+            }
+        }
+#else
         private void SNISSPIData(byte[] receivedBuff, UInt32 receivedLength, byte[] sendBuff, ref UInt32 sendLength)
         {
+            // if we do not have SSPI data coming from server, send over 0's for pointer and length
```

Comment:
This is the original location:
```
if (receivedBuff == null)
{
    // we do not have SSPI data coming from server, so send over 0's for pointer and length
    receivedLength = 0;
}
```
I put the comment back to one line below, which is original location.